x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -67,6 +58,8 @@ const errorHandler = <E extends Error>(
   res: KibanaResponseFactory,
   error: E
 ): IKibanaResponse => {
+  logger.error(error);
+
```

Comment:
Now that you removed the `Boom` error, perhaps introduce logic in this error handler to recognize the new Custom error that also can have a `statusCode` and use that in the in sending an API response. Example:

```
if (error instanceof CustomRequestError {
  return res.customError({
    statusCode: error.statusCode,
    body: error,
  });
}
```

Reply:
FYI - I can see us lifting this error handler method and promote it up so that it can be reused across APIs route handlers

Reply:
Indeed. Where do you suggest we move this so it can be reused? Should we do it in a new PR?

Reply:
it can be a new PR and not part of this. Perhaps the next time we introduce a new API (@parkiino FYI - I know you're working on an new one) we can lift it then